### PR TITLE
chore: use trezor/connect-web and bump

### PR DESCRIPTION
**Pull request by derHowie**
Fixes BX-1333

## What changed (plus any additional context for devs)
We are now importing trezor/connect-web directly and we've bumped the sdk to v9.4.2

## Screen recordings / screenshots
PoW:
- [onboard](https://www.loom.com/share/07282cd1007e4871b080898f752eb76b)
- [sign typed data](https://www.loom.com/share/831d6da817464baa91e78b50a9e2df74)
- [swap](https://www.loom.com/share/39831f072f3e418c96657a86c489a4a1)
- [send](https://www.loom.com/share/5d2f2d5f821b477289c035cecd0a7800)

## What to test
Critical path above and whatever else you think of.

### CODE REVIEW

File: src/core/utils/detectScriptType.ts [hunk 1]
```diff
@@ -9,7 +9,11 @@ export function detectScriptType() {
     if (window.location.pathname.includes('background')) return 'background';
     if (window.location.pathname.includes('contentscript'))
       return 'contentScript';
-    if (window.location.pathname.includes('popup')) return 'popup';
+    if (
+      window.location.pathname.includes('popup') &&
+      !window.location.origin.includes('trezor')
+    )
+      return 'popup';
```
Comment (derHowie) [inline src/core/utils/detectScriptType.ts:16]: Trezor's flow contains a page with 'popup' in the pathname, so we needed to be more specific here.

File: src/entries/popup/App.tsx [hunk 2]
```diff
@@ -66,12 +67,14 @@ export function App() {
...
     // Init trezor once globally
-    window.TrezorConnect?.init({
+    TrezorConnect?.init({
       manifest: {
         email: 'support@rainbow.me',
         appUrl: 'https://rainbow.me',
       },
       lazyLoad: true,
+      transports: ['BridgeTransport', 'WebUsbTransport'],
+      connectSrc: 'https://connect.trezor.io/9/',
```
Comment (derHowie) [inline src/entries/popup/App.tsx:77]: removes the need for a script tag in popup.html

File: src/entries/popup/handlers/trezor.ts [hunk 3]
```diff
@@ -60,8 +56,9 @@ export async function signTransactionFromTrezor(
     }
 
     const nonceHex = BigNumber.from(transaction.nonce).toHexString();
-    const response = await window.TrezorConnect.ethereumSignTransaction({
+    const response = await TrezorConnect.ethereumSignTransaction({
       path,
+      // @ts-expect-error --- unsure why the ts compiler is displaying 'never' for tx type
```
Comment (derHowie) [inline src/entries/popup/handlers/trezor.ts:61]: This has been tested and works fine. Again, would appreciate input on these type issues stemming from deps.

File: src/entries/popup/handlers/wallet.ts [hunk 4]
```diff
@@ -342,14 +343,15 @@ export const importAccountAtIndex = async (
     case 'Trezor':
       {
         const path = getHDPathForVendorAndType(index, 'Trezor');
-        const result = await window.TrezorConnect.ethereumGetAddress({
+        // @ts-expect-error --- the type definitons make no sense for this method.
+        // 'address' is required but it should be optional and only used for validation
+        const result = await TrezorConnect.ethereumGetAddress({
```
Comment (derHowie) [inline src/entries/popup/handlers/wallet.ts:348]: Usage in their docs suggests this type is wrong. https://docs.trezor.io/trezor-suite/packages/connect/methods/ethereumGetAddress.html

Comment (derHowie) [reply src/entries/popup/handlers/wallet.ts:348]: And it works ofc

File: src/entries/popup/pages/messages/useHasEnoughGas.ts [hunk 5]
```diff
@@ -14,10 +16,11 @@ export const useHasEnoughGas = (session: ActiveSession) => {
...
-    selectedGas?.gasFee?.amount || '0',
-    toWei(nativeAsset?.balance?.amount || '0'),
-  );
+  const hasEnough = useMemo(() => {
+    return lessThan(
+      selectedGas?.gasFee?.amount || '0',
+      toWei(nativeAsset?.balance?.amount || '0'),
+    );
+  }, [selectedGas?.gasFee?.amount, nativeAsset?.balance?.amount]);
+  return hasEnough;
```
Comment (derHowie) [inline src/entries/popup/pages/messages/useHasEnoughGas.ts:25]: I saw some pretty intense rerendering going on within our dapp request components: https://www.loom.com/share/f8c651f7282f4beeac3ed9fd5a8652b5

This lessens the rerenders slightly and improved behavior a bit during testing but I think the source of the issue stems from some pretty aggressive gas polling worsened by some inefficient rendering.

Wondering if we should throw this vid into a ticket and revisit `useGas`. Maybe we need to play with debounce or throw `enabled` flags on gas hooks and only enable polling in certain UI states.

**Review by brunobar79 — APPROVED** [review]
LGTM! Much cleaner and happy we were able to get rid of the hosted pkg.
Much easier to update in the future if needed!

File: src/entries/popup/pages/messages/useHasEnoughGas.ts (hunk 5 above)
Comment (brunobar79) [reply src/entries/popup/pages/messages/useHasEnoughGas.ts:25]: +1 on creating a ticket with high priority in the TO DO status

**Comment by brunobar79** [comment]
@derHowie  Can you add a different signTypeData PoW when u get a chance? The uniswap one doesn't apply. Only tokens with approvals that support Permit would use it.  Maybe the easiest is to use the test dapp from e2e

**Comment by BrodyHughes** [comment]
**Okay things tested:**
- Onboarding flow
- Mainnet swaps
- Mainnet sends
- Mainnet eth wrap
- L2 swaps
- L2 sends
- L2 wrap
- Bridging from L2 to mainnet

**On rainbowkit-example.vercel.app:**
- Eth send
- Eth sign txn
- Eth sign typed data

_**All the above worked well!**_

**Existing issues:**
- Went through all existing Trezor tickets on Linear and tried to reproduce the issues described and commented whether it worked or not, and if it was no longer an issue I closed it with a comment.

**New issues:**
- When bridging using a third party dapp (tried on portal.zksync.io/bridge and hop.protocol) ran into an issue with gasLimit
- Checking if this is an existing issue or new with this branch now.

**Comment by DanielSinclair** [comment]
@BrodyHughes Added you as a reviewer if you want to give the final check for this migration

**Review by BrodyHughes — APPROVED** [review]
no changes since i last did a big pass so lgtm.

**Pull request merged**